framework/src/mfem/equation_systems/EquationSystem.C

```diff
@@ -161,35 +156,42 @@ EquationSystem::Init(Moose::MFEM::GridFunctions & gridfunctions,
 }
 
 void
-EquationSystem::ApplyEssentialBCs()
+EquationSystem::ApplyEssentialBC(const std::string & var_name,
+                                 mfem::ParGridFunction & trial_gf,
+                                 mfem::Array<int> & global_ess_markers)
 {
-  _ess_tdof_lists.resize(_test_var_names.size());
-  for (const auto i : index_range(_test_var_names))
+  // Set default value of gridfunction used in essential BC. Values
+  // overwritten in applyEssentialBCs
+  if (_essential_bc_map.Has(var_name))
   {
-    auto test_var_name = _test_var_names.at(i);
-    if (!_essential_bc_map.Has(test_var_name))
-      continue;
-
-    // Set default value of gridfunction used in essential BC. Values
-    // overwritten in applyEssentialBCs
-    mfem::ParGridFunction & trial_gf(*(_var_ess_constraints.at(i)));
-    auto * const pmesh = _test_pfespaces.at(i)->GetParMesh();
-    mooseAssert(pmesh, "parallel mesh is null");
-
-    auto bcs = _essential_bc_map.GetRef(test_var_name);
-    mfem::Array<int> global_ess_markers(pmesh->bdr_attributes.Max());
-    global_ess_markers = 0;
+    auto bcs = _essential_bc_map.GetRef(var_name);
```

Comment:
we're getting a reference but not storing a reference?

Reply:
Corrected